pkg/handlers/nutanix/mutation/machinedetails/inject.go

```diff
@@ -100,10 +100,12 @@ func (h *nutanixMachineDetailsPatchHandler) Mutate(
 			spec.Cluster = nutanixMachineDetailsVar.Cluster
 
 			switch {
-			case nutanixMachineDetailsVar.Image != nil:
+			case nutanixMachineDetailsVar.Image != nil && nutanixMachineDetailsVar.ImageLookup == nil:
```

Comment:
Is this change necessary?
The API validation will prevent from both being set, but even if things change in the future, I think the more specific one`nutanixMachineDetailsVar.Image`, should take higher precedence than `nutanixMachineDetailsVar.ImageLookup`.

Right now if both are somehow set it returns `ErrNoImageOrImageLookupSet`.

Reply:
i found that it was only after adding these changes the e2e tests passed.

> I think the more specific onenutanixMachineDetailsVar.Image, should take higher precedence than nutanixMachineDetailsVar.ImageLookup.

that is the existing logic and it wasn't changed.